Mathlib/CategoryTheory/Category/Twop.lean

```diff
@@ -0,0 +1,220 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+
+! This file was ported from Lean 3 source module category_theory.category.Twop
+! leanprover-community/mathlib commit c8ab806ef73c20cab1d87b5157e43a82c205f28e
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Category.Bipointed
+import Mathlib.Data.TwoPointing
+
+/-!
+# The category of two-pointed types
+
+This defines `Twop`, the category of two-pointed types.
+
+## References
+
+* [nLab, *coalgebra of the real interval*]
+  (https://ncatlab.org/nlab/show/coalgebra+of+the+real+interval)
+-/
+
+
+open CategoryTheory Option
+
+universe u
+
+variable {α β : Type _}
+
+/-- The category of two-pointed types. -/
+structure Twop : Type (u + 1) where
```

Comment:
Not necessary, but perhaps we could take the opportunity to change this to the more readable `TwoP`.